presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -67,6 +67,7 @@
     private final ColumnType columnType;
     private final Optional<String> comment;
     private final List<Subfield> requiredSubfields;
+    private final Optional<Subfield> pushdownSubfield;
```

Comment:
We can reuse `requiredSubfields` instead of introducing a new field. Use `SYNTHESIZED` to distinguish the difference. Let me know if this approach will miss anything.

Reply:
Made the change to use columnType as SYNTHESIZED and use requiredSubfield to store the pushed down Subfield. 

There was one change I had to make in `HiveColumnHandle.withRequiredSubfields` where we have to check if the column is already a pushed down subfield. If it is then don't store the `requiredSubfields` (this gets called from `PushdownSubfields` rule, specifically for the case `arbitrary(x.a).b` where `SubfieldPushdowns` is trying to give an hint to reader that only `b` under column `x.a` is needed. This is a special handling done as part of the a123e5c2a54bfdd7b2ee8860ae62671c146bbdd4.